### PR TITLE
Add example and update readme for Login Plugin

### PR DESCRIPTION
**Pull request by dgodinez-dh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,3 +50,33 @@ t.setAttribute("PluginName", "ExamplePlugin")
 ## Setting an Application Level Plugin
 
 In the config set `Webapi.app.plugins` to your plugin name. This can be a comma separated list to support multiple plugins.
+
+## Login Plugin
+
+A Login Plugin loads before the user log into the Web Client, and it allows UI permissions to be set. See `ExampleLoginPlugin.jsx` for an example.
```
Comment (mofojed) [inline README.md:56]:
```suggestion
A Login Plugin loads before the user logs in to the Web Client, and it allows UI permissions to be set. See `ExampleLoginPlugin.jsx` for an example.
```

File: README.md [hunk 2]
```diff
@@ -50,3 +50,33 @@ t.setAttribute("PluginName", "ExamplePlugin")
...
+The current supported permissions are:
+
+- canUsePanels - Allows the user access to the `Panels` button in the UI.
+- canCreateDashboard - Allows the user to create a new Dashboard.
+- canCreateCodeStudio - Allows the user to create a new Code Studio.
+- canCreateQueryMonitor - Allows the user to open the Query Monitor.
```
Comment (mofojed) [inline README.md:62]: May as well keep the language consistent
```suggestion
- canCreateQueryMonitor - Allows the user to create a new Query Monitor.
```

File: examples/ExampleLoginPlugin.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        canCreateCodeStudio: true,
+        canCreateQueryMonitor: true,
+        canCopy: true,
+        canDownloadCsv: true,
+      };
+      onPermissionOverrides((username, operateAs, groups) => {
+        console.log('[ExampleLoginPlugin]', username, operateAs, groups);
+        return permissionOverrides;
+      });
+    }
```
Comment (mofojed) [inline examples/ExampleLoginPlugin.jsx:24]: Do we have to check if the prop is non-null here? All Login plugins should get this prop. I think this example would be a lot simpler if it was just:
```suggestion
    const { onPermissionOverrides } = this.props;
    onPermissionOverrides((username, operateAs, groups) => {
      console.log('[ExampleLoginPlugin]', username, operateAs, groups);
      return {
        canUsePanels: true,
        canCreateDashboard: true,
        canCreateCodeStudio: true,
        canCreateQueryMonitor: true,
        canCopy: true,
        canDownloadCsv: true,
      };
    });
```

**Review by mofojed — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (dgodinez-dh) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (dgodinez-dh) [reply README.md]: Fixed

File: examples/ExampleLoginPlugin.jsx (hunk 3 above)
Comment (dgodinez-dh) [reply examples/ExampleLoginPlugin.jsx]: Fixed

File: examples/ExampleLoginPlugin.jsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      canCreateDashboard: true,
+      canCreateCodeStudio: true,
+      canCreateQueryMonitor: true,
+      canCopy: true,
+      canDownloadCsv: true,
+    };
+    onPermissionOverrides((username, operateAs, groups) => {
+      console.log("[ExampleLoginPlugin]", username, operateAs, groups);
+      return permissionOverrides;
+    });
```
Comment (mofojed) [inline examples/ExampleLoginPlugin.jsx:22]: I think put the object right in line, no need to create it before:
```suggestion
    const { onPermissionOverrides } = this.props;
    onPermissionOverrides((username, operateAs, groups) => {
      console.log("[ExampleLoginPlugin]", username, operateAs, groups);
      return {
        canUsePanels: true,
        canCreateDashboard: true,
        canCreateCodeStudio: true,
        canCreateQueryMonitor: true,
        canCopy: true,
        canDownloadCsv: true,
      };
    });
```

**Review by mofojed — CHANGES_REQUESTED** [review]

File: examples/ExampleLoginPlugin.jsx (hunk 4 above)
Comment (dgodinez-dh) [reply examples/ExampleLoginPlugin.jsx]: Fixed

**Review by mofojed — APPROVED** [review]

**Pull request merged**